app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -66,13 +67,14 @@ class ProfileListAdapter(
             binding.TxtImg.avatarBackgroundColor = bgColors[item.colorIndex]
 
             binding.profileCard.setOnClickListener {
-                val args = Bundle()
-                args.putParcelable("Profile", item)
-                val navOptions = NavOptions.Builder().setEnterAnim(R.anim.nav_default_enter_anim).setExitAnim(
-                    R.anim.nav_default_exit_anim
-                ).setPopEnterAnim(R.anim.nav_default_pop_enter_anim).setPopExitAnim(R.anim.nav_default_pop_exit_anim)
-                    .build()
-                Navigation.findNavController(it).navigate(R.id.detailsFragment, args, navOptions)
+                fun newInstance() = DetailsFragment().apply {
```

Comment:
You don't need to define a function inside the onClickListener.
Have a look at this article(https://blog.mindorks.com/implementing-dialog-fragment-in-android) and see how it is done.
It is not exactly based on this topic so skip the initial part and see how they define the instance of a fragment.